thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -254,14 +290,46 @@ export default Ember.Controller.extend({
         .reduce((agg, l) => agg.concat(l), []);
 
       aggregatesService.request(context, new Set(offsetUrns));
-      
+
       // scores
       const scoresUrns = aggregatesUrns;
-      
+
       scoresService.request(context, new Set(scoresUrns));
     }
   ),
 
+  /**
+   * Setup observer for context and default selection
+   * May run multiple times while entities are loading.
+   */
+  _setupObserver: Ember.observer(
+    'context',
+    'entities',
+    'scores',
+    'setupMode',
+    function () {
+      const { setupMode } =
+        this.getProperties('setupMode');
+
+      switch (setupMode) {
+        case ROOTCAUSE_SETUP_MODE_NONE:
+          // left blank
+          break;
+
+        case ROOTCAUSE_SETUP_MODE_CONTEXT:
+          this._setupContext();
+          break;
+
+        case ROOTCAUSE_SETUP_MODE_SELECTED:
+          this._setupSelected();
+          break;
+
+        default:
+          throw new Error(`Unknown setup mode '${setupMode}'`);
```

Comment:
Is this error currently getting caught?

Reply:
No. This is a "must not happen" case. We can also ignore it transparently, but may see unexpected behavior later